File: Python Excercise/Part_2.py
# Part 2: Genre Preferences
# 10 users select their favorite movie genres. Some pick the same genre multiple times.

# Write a Python program that:

# Takes 10 strings as input (genres like "Action", "Comedy", "Drama", etc.)
# Adds them to a List (to see all selections)
# Adds them to a Set (to see unique genres)
# Creates a Dictionary with genre name as key and count as value
# Prints all three collections
# Example Input:

# Action, Comedy, Action, Drama, Horror, Action, Comedy, Drama, Thriller, Action
# Example Output:

# List: ['Action', 'Comedy', 'Action', 'Drama', 'Horror', 'Action', 'Comedy', 'Drama', 'Thriller', 'Action']

# Set: {'Action', 'Comedy', 'Drama', 'Horror', 'Thriller'}

# Dictionary: {'Action': 4, 'Comedy': 2, 'Drama': 2, 'Horror': 1, 'Thriller': 1}

# Part 2: Genre Preferences

genres = []

print("Enter 10 movie genres:")

for i in range(10):
    g = input(f"Genre {i+1}: ")
    genres.append(g)

# List (all selections)
genre_list = genres

# Set (unique)
genre_set = set(genres)

# Dictionary (count occurrences)
genre_dict = {}

for g in genres:
    if g in genre_dict:
        genre_dict[g] += 1
    else:
        genre_dict[g] = 1

print("\nList:", genre_list)
print("Set:", genre_set)
print("Dictionary:", genre_dict)

File: javaex/Part_2.java
// Part 2: Ticket Categories
// 10 fans select their ticket preferences. Some pick the same category multiple times.

// Write a Java program that:

// Takes 10 strings as input (ticket types like "VIP", "General", "Standing", etc.)
// Adds them to an ArrayList (to see all selections)
// Adds them to a HashSet (to see unique categories)
// Creates a HashMap with category name as key and count as value
// Prints all three collections
// Example Input:

// VIP, General, VIP, Standing, Seated, VIP, General, Standing, Premium, VIP
// Example Output:

// ArrayList: [VIP, General, VIP, Standing, Seated, VIP, General, Standing, Premium, VIP]

// HashSet: [VIP, General, Standing, Seated, Premium]

// HashMap: {VIP=4, General=2, Standing=2, Seated=1, Premium=1}

import java.util.ArrayList;
import java.util.HashMap;
import java.util.HashSet;
import java.util.Scanner;

public class Part_2 {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        ArrayList<String> list = new ArrayList<>();
        HashSet<String> set = new HashSet<>();
        HashMap<String, Integer> map = new HashMap<>();

        // Taking 10 inputs
        System.out.println("Enter 10 ticket categories:");
        for (int i = 0; i < 10; i++) {
            String ticket = sc.nextLine();
            list.add(ticket);      // Add to ArrayList
            set.add(ticket);       // Add to HashSet

            // Add to HashMap (count occurrences)
            map.put(ticket, map.getOrDefault(ticket, 0) + 1);
        }

        // Printing all collections
        System.out.println("\nArrayList: " + list);
        System.out.println("HashSet: " + set);
        System.out.println("HashMap: " + map);

        sc.close();
    }    
}
